### PR TITLE
feat(client/kotlin): add Favorite Resources menu

### PR DESCRIPTION
**Pull request by ReactorScram**
It's missing a couple pieces, see the tasklist

image

Refs #5123

```[tasklist]
- [x] Add `Add to Favorites` and `Remove from Favorites` buttons
- [x] Update Changelog
- [x] Load and save Favorites from `SharedPreferences`
- [x] Wire up `onClick` events
- [x] Hide and show Resources in the menu based on whether they're favorited
- [x] Hide tabs if nothing is favorited
- [x] Tab icons
- [ ] Make the "Reset Settings" button also reset Favorites
- [ ] Change the "Add to Favorites" and "Remove from Favorites" to a checkbox or star or something cool
```

### CODE REVIEW

File: website/src/components/Changelog/Android.tsx
Comment (ReactorScram) [inline website/src/components/Changelog/Android.tsx:4]: Alphabetized

Comment (thomaseizinger) [reply website/src/components/Changelog/Android.tsx:4]: Can we enforce this with a formatter?

File: kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt
Comment (thomaseizinger) [inline kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt]: I think this should be private and we expose a getter and setter instead? AFAIK that is the Java / Kt convention. We do the same for `resourcesLiveData`.

File: kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionActivity.kt
Comment (thomaseizinger) [inline kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionActivity.kt]: UIs are notoriously hard to test. View-models help with that because they can act as a place where you put all your logic in what should get displayed. Consequently, the actual code that changes the UI is so "stupid" that you don't need to test it.

Instead of this conditional logic here, I'd suggest to create functions on `viewModel` that compute you the things that you need:

- A getter for `showOnlyFavourites`
- A getter for the tab visibility
- A getter for the tab index
- A getter for which resources are sent to the `resourcesAdapter`

File: kotlin/android/app/src/main/res/layout/activity_session.xml
Comment (thomaseizinger) [inline kotlin/android/app/src/main/res/layout/activity_session.xml:76]: This isn't a blocking concern but more of a question.

I don't really understand the choice of tabs here. To me, tabs are useful to separate unrelated entities, like Messages, Calls and Status in Whatsapp. 

Favourite and non-favourite resources are still resources. Why split them into different tabs? In Fastmail for example, I can "Pin" emails (which is basically like a favourite). Those are sticky at the top of all my emails and the rest is sorted by "Recevied" date.

File: kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt
Comment (thomaseizinger) [inline kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt]: I don't know much about Kotlin but can't we initialise this with an empty set like `_resourcesLiveData`? That could save us a few lines of code maybe?

**Review by thomaseizinger — APPROVED** [review]
Nice work! I left some ideas and a question.

**Review by Patticatti — APPROVED** [review]
Looks nice - maybe we can start leaning into theme colours eventually

File: kotlin/android/app/src/main/res/layout/activity_session.xml
Comment (ReactorScram) [reply kotlin/android/app/src/main/res/layout/activity_session.xml:76]: Absolutely no reason, I just had the tabs half-way implemented and I forgot what the conclusion was for UX.

@Patticatti If the pinned / sticky favorites sounds good, I'll change it to that ~~before merging~~ later on. Your call as UX person

File: website/src/components/Changelog/Android.tsx
Comment (ReactorScram) [reply website/src/components/Changelog/Android.tsx:4]: I'm not aware of one, but probably

File: kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt
Comment (ReactorScram) [reply kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt]: Done. It does change the Elvis operator into just `!!` but other than that it's still optional. :/

**Pull request merged**
